autoload/firenvimft.vim

```diff
@@ -1,7 +1,7 @@
 
 let s:patterns_to_ft = {
         \ '/github.com_.*\.txt$': 'markdown',
-        \ '/reddit.com_.*\.txt$': 'markdown',
+        \ '/www.reddit.com_.*\.txt$': 'markdown',
```

Comment:
Wouldn't something like
```suggestion
        \ '/\(\w\+.\)*reddit.com_.*\.txt$': 'markdown',
```
Would be better as it'd match both the `www.` and the `old.` subdomains.

Reply:
oh, I wasn't aware of the `old` subdomain, I'm not an avid redditor :)

but what you have written doesn't quite make sense to me. What kind of regex is that, it seems to revert escaping? Does a dot now match a plain dot or any character?
And that star should be a question mark.

Reply:
>What kind of regex is that, it seems to revert escaping?

I think vim operators use `magic` by default which reverts the usual meaning of special characters like `(`, `+` and `)`, so if you want literal characters you need to escape them. We don't need a double backslash because single-quoted strings don't interpret backslash characters.

>Does a dot now match a plain dot or any character?

I think this matches any character, so we'd need to escape the dots. Good catch :)

>And that star should be a question mark.

I think the star is useful as it'll match multiple nested subdomains.

Reply:
a yes, then the star does make sense. Didn't think of that.